kore/src/Kore/ProofState.hs

```diff
@@ -0,0 +1,63 @@
+{-|
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+-}
+module Kore.ProofState
```

Comment:
I wanted to move `ProofState` and `Goal` in the `Strategies` package, but I vaguely remembered (so it is entirely possible that I'm wrong) that they were in a separate package, and we moved them at the top level, and I wasn't sure why.

Other than that, yes, I would move them down in the hierarchy.

Reply:
We moved them up because they were in `Kore.AllPath` before, but they obviously apply to `OnePath` or `AllPath` just as well. At that time, `OnePath` was at the top level, so they had nowhere to go but up to the top level, too. I think it's fine to move them down into `Kore.Strategies`, but if you want to merge this pull request as-is that's fine, too.

Reply:
Moved it.